.github/workflows/algolia/src/uploadToAlgolia.ts

```diff
@@ -1,7 +1,9 @@
-import { generateAlgoliaJson, uploadAlgoliaData } from ".";
+import { generateAlgoliaJson, uploadAlgoliaData, deleteDocsIndex } from ".";
 
 const main = async () => {
   const records = generateAlgoliaJson();
+  await deleteDocsIndex();
+  console.log("Old Index Deleted.");
   await uploadAlgoliaData(records);
```

Comment:
Looks like we create two Algolia client instances (in `deleteDocsIndex` and `uploadAlgoliaData`). We can probably get away with just one method (e.g. `syncAlgoliaIndex`) which receives the new ‘truth’ (records) and either

-  deletes all old & uploads all new records
- diffs records with the existing search and makes small atomic requests (e.g. delete 3 records and send (update/create) 5)

Reply:
I believe diffing would be an even more expensive operation, since it would involve:
1. Checking if a record exists. If not, delete the record.
2. If record exists, check every key for a change. This would become more complex for nested values.

Overall, this would result in a O(num_records * num_keys) operation. At least with the above steps.

Is there any alternate algo you had in mind for this. Should we even worry about rate limits, since the current count of all docs + glossary pages is at ~231.

I do understand the concern about using `delete` over `deleteBy` because the doc does state this:
>This method should only be used if you can’t get the list of objectIDs you want to delete

Reply:
We could at least only delete files that have been moved or delete. Then we are not wiping the index each time - seems like this could introduce problems with search between the deletion and the upload.

Reply:
@CiaranMn this is my very naive approach to how a diff-based approach would work. Do you have any thoughts on this?